syne_tune/optimizer/schedulers/searchers/gp_fifo_searcher.py

```diff
@@ -449,6 +454,36 @@ class GPFIFOSearcher(ModelBasedSearcher):
         because criterion is only used internally. Also note that criterion
         data is always normalized to mean 0, variance 1 before fitted with a
         GP.
+    transfer_learning_task_attr : str (optional)
+        Used to support transfer HPO, where the state contains observed data
+        from several tasks, one of which is the active one. To this end,
+        `configspace` must contain a categorical parameter of name
+        `transfer_learning_task_attr`, whose range are all task IDs. Also,
+        `transfer_learning_active_task` must denote the active task, and
+        `transfer_learning_active_config_space` is used as
+        `active_config_space` argument in :class:`HyperparameterRanges` (this
+        allows us to use a narrower search space for the active task than for
+        the union of all tasks (`configspace` must be that).
+    transfer_learning_active_task : str (optional)
+        See `transfer_learning_task_attr`.
+    transfer_learning_active_config_space : Dict (optional)
+        See `transfer_learning_task_attr`. If not given, `configspace` is the
+        search space for the active task as well. This active config space need
+        not contain the `transfer_learning_task_attr` parameter. In fact, this
+        parameter is set to a categorical with `transfer_learning_active_task`
+        as single value, so that new configs are chosen for the active task
+        only.
+    transfer_learning_model : str (optional)
+        See `transfer_learning_task_attr`. Specifies the surrogate model to be
+        used for transfer lerning:
+        - 'matern52_product': Kernel is product of Matern 5/2 (not ARD) on
+            `transfer_learning_task_attr` and Matern 5/2 (ARD) on the rest.
+            Assumes that data from same task are more closely related than
+            data from different tasks
+        - 'matern52_same': Kernel is Matern 5/2 (ARD) on the rest of the
+            variables, `transfer_learning_task_attr` is ignored. Assumes
+            that data from all tasks can be merged together
```

Comment:
What happens if you pass a large number of observations? Perhaps not for this PR but could we have a simple baseline to trim down to a reasonable number (for instance by sampling or grabing top ones

Reply:
True, this particular way of doing warm-starting does not scale well.